src/sequence_manager.hpp

```diff
@@ -47,9 +55,24 @@ class SequenceManager {
 
 public:
     SequenceManager() = default;
-    SequenceManager(uint32_t timeout, uint32_t maxSequenceNumber) :
+    SequenceManager(uint32_t timeout, uint32_t maxSequenceNumber, std::string modelName) :
```

Comment:
If you're adding modelName to sequence manager fields for logging purposes you could also add model version here.
Or even pass it as std::pair to have less parameters and track the together.